Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/AndroidModule.kt

```diff
@@ -63,4 +65,7 @@ class AndroidModule {
     @Provides
     @Singleton
     fun activityManager(@AppContext context: Context): ActivityManager = context.getSystemService()!!
+
+    @Provides
+    fun stringProvider(@AppContext context: Context): StringProvider = ContextStringProvider(context)
```

Comment:
I'm not a fan using the application context, isn't it better to use the activity/fragment/view context to resolve the strings, and resolve them as close as possible to UI?

We have a similar pattern in the app, see `LazyString`, which reverses the responsibility, instead of using a provider to resolve using the application context, we create a getter where the caller has to input their own context to get the string.

Reply:
Something maybe like

```kotlin
fun Int.toResolvingString(vararg formatArgs: Any): LazyString = object : LazyString {
    override fun get(context: Context): String = context.getString(this@toResolvingString, *formatArgs)
}

Item(
    R.string.my_string_resource.toResolvingString()
)

````

Reply:
This is used to resolve accessibility strings for the list items. Strings need to be resolved for each element, depending on states only known in the viewmodel. I did not want to move that much logic into the fragment. 
I saw the LazyString but did not find it useful in this particular case.
Is there a problem with using the app context?

Reply:
I can use an assisted inject with the fragment's context

Reply:
> Strings need to be resolved for each element, depending on states only known in the viewmodel. I did not want to move that much logic into the fragment.

The logic would still be in the viewmodel, just the resolving would be moved to the fragment. It would look like:

```kotlin
SelectableItem(
    true,
    contactDiaryLocation,
    SELECTED_CONTENT_DESCRIPTION.toResolvingString(contactDiaryLocation.locationName),
    (...)
)

// Later in UI
it.contentDescription = item.onClickDescription.get(context)
```

> Is there a problem with using the app context?

It's good practice to not use the app context for UI related operations. The internal call will perform `context.resources.getString(...)` and `resources` can be different between app context and activity context, resources is affected by theming and other options that can be different per activity.

Reply:
> I can use an assisted inject with the fragment's context

That would be okay too. Not sure about testing. Up to you.

Reply:
I added what you proposed